### DISCUSSION TITLE
storing and getting structs

### DISCUSSION THREAD

**Discussion by enderNakamoto** [aptos-labs/aptos-developer-discussions#445] [Questions] [answered]
### Discord user ID

enderwiggins

### Describe your question in detail.

Hi, this example - https://learn.aptoslabs.com/en/tutorials/aptos-101/move-language/modules?workshop=web3-to-aptos has an inconsistency, but more likely I do not understand something: 

Here's the code:
```
module aptogotchi::main {

	  struct Config has copy, drop {
	  	admin: address
	  }

      // Init function runs once when the module is published for the first time.
	  fun init_module(caller: &signer) {
        let caller_addr = address_of(caller);

        // move_to is a function to move the struct to the global storage
        move_to(caller, Config { admin: caller_addr });
    }

    #[view]
    public fun get_admin(): address acquires Config {
        // @aptogotchi is the address of the module
        let config = borrow_global<Config>(@aptogotchi);

        config.admin
    }
}

```

### Questions

1.   if the struct is stored in caller - move_to(caller, Config { admin: caller_addr });, how is it retreived from module address? let config = borrow_global<Config>(@aptogotchi);. Are caller/module address same? in that case, isn't admin just aptogotchi??

2. why does Config has copy, drop, shouldn't it have key as it is stored in an address?

**Comment by gregnazario** [comment, marked as answer]
> if the struct is stored in caller - move_to(caller, Config { admin: caller_addr });, how is it retreived from module address? let config = borrow_global(@aptogotchi);. Are caller/module address same? in that case, isn't admin just aptogotchi??

`init_module` is called on publishing of a module, with the publisher's signer.  The publisher's address is the same address as the module (in this case `aptogotchi`), so they are the same address.

And yes, in this case admin is `@aptogotchi`, but it could be changed to something else.

> why does Config has copy, drop, shouldn't it have key as it is stored in an address?

This one should most certainly have `key`, I'm guessing it was inconsistent cc @0xaptosj

**Reply by 0x-j** [reply]
yeah it should have `key` capability. FIxed and should be live soon
